ludwig/schema/encoders/utils.py

```diff
@@ -36,6 +38,33 @@ def get_encoder_classes(feature: str):
     return encoder_config_registry[feature]
 
 
+@DeveloperAPI
+def get_encoder_descriptions(feature_type: str):
+    """This function returns a dictionary of encoder descriptions available at the type selection.
+
+    The process works as follows - 1) Get a dictionary of valid encoders from the encoder config registry,
+    but inverse the key/value pairs since we need to index `valid_encoders` later with an altered version
+    of the encoder config class name. 2) Loop through Encoder Metadata entries, if a metadata entry has an
+    encoder name that matches a valid encoder, add the description metadata to the output dictionary.
+
+    Args:
+        feature_type (str): The feature type to get encoder descriptions for
+    Returns:
+         dict: A dictionary of encoder descriptions
+    """
+    output = {}
+    valid_encoders = {
+        cls.module_name() if hasattr(cls, "module_name") else None: registered_name
+        for registered_name, cls in get_encoder_classes(feature_type).items()
+    }
+
+    for k, v in ENCODER_METADATA.items():
+        if k in valid_encoders.keys():
```

Comment:
With above suggestion about making valid_encoders a set or list
```suggestion
        if k in valid_encoders:
```

Reply:
see above comment.